hotham/src/systems/hands.rs

```diff
@@ -59,55 +66,48 @@ pub fn hands_system_inner(world: &mut World, input_context: &InputContext) {
     }
 }
 
-/// Convenience function to add a Hand and corresponding Mesh to the world
+/// Convenience function to add a Hand, Collider and corresponding Mesh to the world
 pub fn add_hand(
     models: &std::collections::HashMap<String, World>,
     handedness: Handedness,
     world: &mut World,
     physics_context: &mut PhysicsContext,
 ) {
-    let model_name = match handedness {
-        Handedness::Left => "Left Hand",
-        Handedness::Right => "Right Hand",
+    let (hand_component, model_name) = match handedness {
+        Handedness::Left => (Hand::left(), "Left Hand"),
+        Handedness::Right => (Hand::right(), "Right Hand"),
     };
-    let hand = add_model_to_world(model_name, models, world, physics_context, None).unwrap();
-    {
-        // Add a hand component
-        world
-            .insert_one(
-                hand,
-                Hand {
-                    grip_value: 0.,
-                    handedness,
-                    grabbed_entity: None,
-                },
-            )
-            .unwrap();
 
-        // Modify the animation controller
-        let mut animation_controller = world.get::<&mut AnimationController>(hand).unwrap();
-        animation_controller.blend_from = 0;
-        animation_controller.blend_to = 1;
-        drop(animation_controller);
-
-        // Give it a collider and rigid-body
-        let collider = ColliderBuilder::capsule_y(0.05, 0.02)
-            .sensor(true)
-            .active_collision_types(ActiveCollisionTypes::all())
-            .active_events(ActiveEvents::COLLISION_EVENTS)
-            .build();
-        let rigid_body = RigidBodyBuilder::new(RigidBodyType::KinematicPositionBased).build();
-        let components = physics_context.create_rigid_body_and_collider(hand, rigid_body, collider);
-        world.insert(hand, components).unwrap();
-    }
+    // Spawn the hand
+    let hand_entity = add_model_to_world(model_name, models, world, physics_context, None).unwrap();
+
+    // Modify the animation controller
+    let mut animation_controller = world.get::<&mut AnimationController>(hand_entity).unwrap();
+    animation_controller.blend_from = 0;
+    animation_controller.blend_to = 1;
+    drop(animation_controller);
+
+    // Give it a collider and rigid-body
```

Comment:
No rigid body anymore.

Reply:
Ah, good pickup. Also forgot to set the `user_data`!